### ISSUE TITLE
Docked app is unusable after a reboot

### ISSUE THREAD

**Issue by igmar** [mattermost/desktop#411]
I confirm (by marking "x" in the [ ] below): 

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [x] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md). 

---

**Summary**
A docked app is unusable after a reboot

**Steps to reproduce**

OSX 10.12.2
Desktop app 3.5.0
Server 3.5.1 

Start Mattermost app -> Keep in Dock -> reboot
The app turns into a question mark, and needs to be readded

**Expected behavior**

No need to add again.

**Comment by jasonblais** [comment]
@yuya-oc or @magicmonty would you be able to confirm the given reproduction steps on your Mac?

**Comment by jasonblais** [comment]
and thanks @igmar for the bug report :)

**Comment by yuya-oc** [comment]
I could reproduce the issue when I launched the app from download folder. But I could not reproduce the issue when I launched the app from `~/Application`.

1. Open Mattermost.app which is located in `~/Download`.
    - probably any folder excepting Application folder.
2. Add to Dock via context menu.
3. Reboot Mac.

@igmar Would you tell us where you located Mattermost.app?

**Comment by igmar** [comment]
On both tests, Mattermost.app is located in /Applications/

**Comment by yuya-oc** [comment]
Hmm...I could not reproduce the issue also in `/Applications`.

Could you show more detailed steps about the way to add to Dock?

**Comment by colintheshots** [comment]
I'm seeing the same behavior. App is installed at /Applications/Mattermost.app.

**Comment by jasonblais** [comment]
Any updates on here?

**Comment by igmar** [comment]
We stopped using Mattermost, it wasn't suitable for production usage.

**Comment by jasonblais** [comment]
Sorry to hear that, but also understand -- what made you move away from Mattermost, if I may ask?

**Comment by yuya-oc** [comment]
I think the problem was solved probably by upgrading Electron. I tested with the latest artifact.

https://circleci.com/gh/mattermost/desktop/930#artifacts

**Comment by jasonblais** [comment]
@igmar @colintheshots would you like to try downloading test build from https://circleci.com/gh/mattermost/desktop/930#artifacts and see if the issue is resolved for you?

**Comment by igmar** [comment]
@jasonblais Basically, to many issues. Apps not working properly, events not getting noticed. We unleashed our QA department on it for 2 hours, and at a point they just stopped reporting issues.

While that might be fine for an unpaid product, it's not if it's a paid product.

**Comment by jasonblais** [comment]
Thanks @igmar for the feedback, much appreciate it. We respect your decision and hope the platform you chose meets your needs.

If you'd be open to, we'd love to see issue reports your QA department filled. Of course understand if they're confidential. We have moved the Mattermost server back to monthly releases due to increased community activity, allowing us to fix bugs and quality issues sooner. We also released a beta version [of our new mobile apps](https://github.com/mattermost/mattermost-mobile), as the old apps we're not well-performant particularly on Android.

Thanks again for all your feedback,
